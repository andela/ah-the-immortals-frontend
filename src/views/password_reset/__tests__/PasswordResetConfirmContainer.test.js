import React from 'react';
import { Router } from 'react-router-dom';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { createBrowserHistory } from 'history';
import { Form } from 'react-bootstrap';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import toJson from 'enzyme-to-json';
import PasswordResetConfirm from '../PasswordResetConfirmContainer';

describe('Tests PasswordResetConfirmContainer', () => {
  const store = configureStore([thunk])({
    confirmresetpassword: { message: {}, errors: {} }
  });
  const props = {
    match: {
      params: {
        token: 'tcvygbuhijn6789'
      }
    },
    showModal: jest.fn(),
    history: {push: jest.fn()}
  };
  const history = createBrowserHistory();
  const wrapper = mount(
    <Router history={history}>
      <Provider store={store}>
        <PasswordResetConfirm {...props} />
      </Provider>
    </Router>
  );
  it('Test mounting of component', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('Tests input change in form', () => {
    const form = wrapper.find(Form);
    const input = form.find('FormControl[name="password"]');
    input.simulate('change', {
      preventDefault: jest.fn(),
      target: {
        value: 'issamwangi@gmail.com',
        password: 'MainA9176!'
      }
    });
    expect(input.length).toEqual(1);
  });
  it('handles submit in form', () => {
    const PasswordChangeBtn = wrapper.findWhere(n => n.type() === 'button' && n.contains('Submit'));
    PasswordChangeBtn.simulate('submit', {
      preventDefault: jest.fn()
    });
    expect(PasswordChangeBtn.length).toEqual(1);
  });
});
