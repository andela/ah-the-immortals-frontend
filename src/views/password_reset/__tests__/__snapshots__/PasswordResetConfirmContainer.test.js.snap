// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests PasswordResetConfirmContainer Test mounting of component 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Connect(PasswordResetConfirm)
      history={
        Object {
          "push": [MockFunction],
        }
      }
      match={
        Object {
          "params": Object {
            "token": "tcvygbuhijn6789",
          },
        }
      }
      showModal={[MockFunction]}
    >
      <PasswordResetConfirm
        confirmPasswordResetAction={[Function]}
        errors={Object {}}
        history={
          Object {
            "push": [MockFunction],
          }
        }
        match={
          Object {
            "params": Object {
              "token": "tcvygbuhijn6789",
            },
          }
        }
        message={Object {}}
        showModal={[MockFunction]}
      >
        <div>
          <Container
            className="reset"
            fluid={false}
          >
            <div
              className="reset container"
            >
              <div>
                <h1
                  className="resetmessage"
                >
                  Create your new password.
                </h1>
                <p>
                  Password must include at least 8 characters including at least 1 number, 1 special character,
                  <br />
                  and atleast two uppercase letters.
                </p>
                <br />
              </div>
              <Form
                inline={false}
                onSubmit={[Function]}
              >
                <form
                  className=""
                  onSubmit={[Function]}
                >
                  <FormControl
                    className="confirm-input passwordStyle"
                    id="password"
                    name="password"
                    onChange={[Function]}
                    placeholder="Enter your new password"
                    required={true}
                    type="password"
                  >
                    <input
                      className="confirm-input passwordStyle form-control"
                      id="password"
                      name="password"
                      onChange={[Function]}
                      placeholder="Enter your new password"
                      required={true}
                      type="password"
                    />
                  </FormControl>
                  <FormControl
                    className="confirm-input passwordStyle"
                    id="password_confirm"
                    name="password_confirm"
                    onChange={[Function]}
                    placeholder="Confirm your new password"
                    required={true}
                    type="password"
                  >
                    <input
                      className="confirm-input passwordStyle form-control"
                      id="password_confirm"
                      name="password_confirm"
                      onChange={[Function]}
                      placeholder="Confirm your new password"
                      required={true}
                      type="password"
                    />
                  </FormControl>
                  <div
                    className="reset-buttons"
                  >
                    <Button
                      active={false}
                      className="reset-buttons"
                      disabled={true}
                      type="submit"
                      variant="primary"
                    >
                      <button
                        className="reset-buttons btn btn-primary"
                        disabled={true}
                        type="submit"
                      >
                        Submit
                      </button>
                    </Button>
                  </div>
                </form>
              </Form>
            </div>
          </Container>
        </div>
      </PasswordResetConfirm>
    </Connect(PasswordResetConfirm)>
  </Provider>
</Router>
`;
